x-pack/legacy/plugins/snapshot_restore/public/app/services/http/use_request.ts

```diff
@@ -61,20 +60,34 @@ export const useRequest = ({
   body,
   interval,
   initialData,
-  timeout,
   uimActionType,
 }: UseRequest) => {
+  // Main states for tracking request status and data
   const [error, setError] = useState<null | any>(null);
   const [loading, setLoading] = useState<boolean>(true);
   const [data, setData] = useState<any>(initialData);
 
+  // States for tracking polling
+  const [polling, setPolling] = useState<boolean>(false);
+  const [currentInterval, setCurrentInterval] = useState<UseRequest['interval']>(interval);
+  const intervalRequest = useRef<any>(null);
+  const isFirstRequest = useRef<boolean>(true);
+
   // Tied to every render and bound to each request.
   let isOutdatedRequest = false;
 
   const request = async () => {
-    setError(null);
-    setData(initialData);
-    setLoading(true);
+    const isPollRequest = currentInterval && !isFirstRequest.current;
+
+    // Don't reset main error/loading states if we are doing polling
```

Comment:
Can we also explain the "why" from a UX perspective?

```js
// Don't reset main error/loading states if we are doing polling because we want
// the reload to be transparent to the user (e.g. no spinner).
```

Reply:
EDIT: I left another comment below which may make this moot.